pkg/controller/tas/topology_ungater.go

```diff
@@ -403,14 +402,14 @@ func readRanksIfAvailable(log logr.Logger,
 
 func determineRanksLookup(pod *corev1.Pod) (*string, *replicatedJobsInfo) {
 	// Check if this is JobSet
-	if jobCount, _ := readIntFromLabel(pod, jobset.ReplicatedJobReplicas); jobCount != nil {
+	if jobCount, _ := utilpod.ReadUIntFromLabel(pod, jobset.ReplicatedJobReplicas); jobCount != nil {
 		return ptr.To(batchv1.JobCompletionIndexAnnotation), &replicatedJobsInfo{
 			jobIndexLabel: jobset.JobIndexKey,
 			replicasCount: *jobCount,
 		}
 	}
 	// Check if this is Pod group
-	if jobCount, _ := readIntFromLabel(pod, kueuealpha.PodGroupPodIndexLabel); jobCount != nil {
+	if jobCount, _ := utilpod.ReadUIntFromLabel(pod, kueuealpha.PodGroupPodIndexLabel); jobCount != nil {
```

Comment:
Oh, I just realized this. I think it is better to follow the pattern as below 
`if _, found := pod.Labels[batchv1.PodGroupPodIndexLabel]; found {` - the `jobCount` does not make sense in this context. The lookup only aims to check if the PodGroup is used. We will parse later on anyway.

It is not related to this PR, so please follow up.  cc @PBundyra